sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3994,10 +3994,45 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.markAsAnalyzed(AnalysisContext.get)
       case c: KeepAnalyzedQuery if c.resolved =>
         c.storeAnalyzedQuery()
-      case p: LogicalPlan if p.resolved =>
-        p.transformAllExpressionsWithPruning(_.containsPattern(ANALYSIS_AWARE_EXPRESSION)) {
-          case e: AnalysisAwareExpression[_] => e.markAsAnalyzed()
+      case c: CreateTable if c.resolved =>
+        c.copy(columns = storeDefaultForColumnDefinition(c.columns))
+      case r: ReplaceTable if r.resolved =>
+        r.copy(columns = storeDefaultForColumnDefinition(r.columns))
+      case a: AddColumns =>
+        // Wrap analysis errors for default values in a more user-friendly message.
+        a.columnsToAdd.foreach { c =>
```

Comment:
note:  This is to throw the previously expected exception (defaultValueUnresolved).  Tried to move it to CheckAnalysis, but its a bit too late and the more generic Unresolved exception is already thrown there if any Expression is still unresolved, before it gets to the checkAlterTable part.

Reply:
My gut tells me to look for another way and keep this rule generic, as it was before. Let me think.

Reply:
I updated it to a better place